pkg/fileio/file_io.go

```diff
@@ -0,0 +1,61 @@
+package fileio
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"text/template"
+)
+
+func WriteFile(filename string, contents string, templateData any) error {
+	if templateData == nil {
+		if err := os.WriteFile(filename, []byte(contents), os.ModePerm); err != nil {
+			return fmt.Errorf("writing file: %w", err)
+		}
+
+		return nil
+	}
+
+	tmpl, err := template.New(filename).Parse(contents)
+	if err != nil {
+		return fmt.Errorf("parsing template: %w", err)
+	}
+
+	file, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("creating file: %w", err)
```

Comment:
If you do decide to add the filename to the previous error message, for consistency this should be updated too.